rfcs/0019-storing-client-data.md

```diff
@@ -0,0 +1,484 @@
+
+The positive effect of having an API process is that it allows other developers to make modifications to queries without having to change the inner works of the daemon or the archive process. For example,this modification would enable users to query data using RPC serialization or Haxl rather than using GraphQL. This also gives us the opprotunity to use services such as Hasura that converts SQL schemas to Graphql schemas to make writing API commands very concise.
+
+The API process will also make subscription to changes in the daemon and changes in the database. The various subscriptions will be discussed in a later part of this RFC.
+
+The following section will describe SQL queries to make complicated queries.
+
+## GraphQL commands
+
+The notion document below discusses the GraphQL commands proposed in this new architecture. It succinctly discusses how it will evolve from the current GraphQL commands. It also discusses which components of the new architecture are involved in making these GraphQL queries. It also shows some sample code on complicated queries such as `transaction_status`. The advantage of this architecutre is that there are no GraphQL queries that depend on both the daemon and the database simultaneously:
+
+https://www.notion.so/codaprotocol/Proposed-Graphql-Commands-57a434c59bf24b649d8df7b10befa2a0
+
+A HTML rendered version of the notion document can be found [here](../docs/res/graphql_diff.html).
+
+## Consistency Issues
+
+As mentioned in the [Running Bootstrap section](#bootstrap), we would not be able to fully determine the `consensus_status` of the block that were in the transition_frontier. Worse, we would not be able to determine which blocks became snarked.
+
+We can have a third-party archived node tell a client the consensus_state of the blocks. Another option that we have is that we can create a P2P RPC call where peers can give a client a Merkle list proof of a state_hash and all the state_hashes of the blocks on top of the state_hash. The Merkle list should be length `K`, the number of block confirmations to gaurantee finality with a very high probability. If the Merkle list proof is size `K`, then the block should be fully confirmed. Otherwise, the block failed.
+
+# Rationale and Alternatives
+
+- Makes it easy to perform difficult queries for our protocol
+- Makes the client code more module
+- The current implementation takes a lot of work and it is quite limited for a certain use case.
+- Other alternatives are discussed in the previous commits of this [RFC](https://github.com/CodaProtocol/coda/pull/2901)
+- In an ideal world where we have an infinite amount of time, we would implement our own native graph database leverage the `transition_frontier` as an in-memory cache. Namely, we would have all the blocks in the `transition_frontier` to be in the in-memory cache along with nodes that have a 1 to 2 degree of seperation
+
```

Comment:
I feel like another alternative is integrating the persistence system with the archival system in some way, or at least describing how they're related since they both involve saving transition frontier information to disk.

I think one or two reasons have been mentioned in person why this shouldn't be the solution we take but I always forget so it could be nice to call it out.